duckduckgo/macos-browser
# Send VPN system extension crashes to Sentry

Task/Issue URL: https://app.asana.com/0/0/1206227536353331/f
Tech Design URL:
CC:

**Description**:

This PR updates the crash report parser to read and send system extension crashes.

It's marked as Do Not Merge until the symbolication on the backend is functional. I've submitted a crash report already and pinged the backend team for help. For now though, the code can still be reviewed.

**Steps to test this PR**:
1. First, we need to remove the DEBUG checks that prevent the crash report runner from executing in debug mode: update the `checkForNewReports` function to no longer check for `!DEBUG`
2. Next, generate a system extension crash using [these instructions](https://app.asana.com/0/1202500774821704/1206227278256234/f)
3. Finally, run the app and make sure that it shows a modal window with the contents of the crash. Note that Sentry does not yet symbolicate these crashes - it won't hurt anything if you choose to send it, but I would suggest clicking Cancel instead just so that we don't have unsymbolicated crashes crowding up the place



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
